packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/AbstractStaticNativeLibraryFeature.kt

```diff
@@ -46,28 +136,85 @@ internal abstract class AbstractStaticNativeLibraryFeature : NativeLibraryFeatur
     }
   }
 
-  /**
-   * TBD.
-   */
-  abstract fun nativeLibs(access: BeforeAnalysisAccess): List<NativeLibInfo?>
+  public abstract fun nativeLibs(access: BeforeAnalysisAccess): List<NativeLibInfo?>
+
+  public open fun unpackNatives(access: BeforeAnalysisAccess) {
+    // native libraries to unpack at build time
+  }
+
+  override fun getRequiredFeatures(): MutableList<Class<out Feature>> {
+    return mutableListOf(
+      JNIFeature::class.java,
+    )
+  }
 
   override fun beforeAnalysis(access: BeforeAnalysisAccess) {
+    unpackNatives(access)
     super.beforeAnalysis(access)
 
     nativeLibs(access).forEach {
       if (it == null) return@forEach  // not supported on this platform
 
       // register lib
-      NativeLibrarySupport.singleton().preregisterUninitializedBuiltinLibrary(it.name)
+      val libSupport = NativeLibrarySupport.singleton()
+      val platformLibs = PlatformNativeLibrarySupport.singleton()
+
+      if (it.builtin) {
+        libSupport.preregisterUninitializedBuiltinLibrary(it.linkName)
+
+//        val libraryInit = NativeLibrarySupport::class.java.getDeclaredField("libraryInitializer")
+//        libraryInit.isAccessible = true
+//        val knownLibrariesField = NativeLibrarySupport::class.java.getDeclaredField("knownLibraries")
+//        val libraryInitializer = libraryInit.get(libSupport) as JNILibraryInitializer
+//        knownLibrariesField.isAccessible = true
+//
+//        libraryInitializer.initialize(
+//          PlatformNativeLibrarySupport.singleton().createLibrary(it.linkName, true),
+//        )
+//        val knownLibraries = knownLibrariesField.get(libSupport) as MutableList<NativeLibrary>
+//        libSupport.preregisterUninitializedBuiltinLibrary(it.linkName)
+//        val target = knownLibraries.find { candidate -> candidate.canonicalIdentifier == it.linkName }
+//
+//        val loadedField = NativeLibrary::class.java.getDeclaredField("isLoaded")
+//        loadedField.isAccessible = true
+//        loadedField.setBoolean(target, true)
+      }
       if (it.registerPrefix) {
-        PlatformNativeLibrarySupport.singleton().addBuiltinPkgNativePrefix(it.prefix)
+        it.prefix.forEach { prefix ->
+          platformLibs.addBuiltinPkgNativePrefix(prefix.replace(".", "_"))
+        }
```

Comment:
registers JNI symbols on the JVM side, like `Java_dev_elide_my_cool_Symbol_here`